docs/extras/headers.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Headers
+---
+# Headers Extra
+
+This extra implements the [RFC-8288](https://tools.ietf.org/html/rfc8288) compilant http response headers (and other helpers) useful for API pagination.
+
+- It removes the need for an extra dependency
+- No need to learn yet another interface
+- It saves quite a lot of memory and CPU
+- It works with any type of collection and/or `pagy_*` constructor (even `pagy_countless`)
+- It offers more explicit flexibility and simplicity
+
+## Synopsis
+
+See [extras](../extras.md) for general usage info.
+
+In the `pagy.rb` initializer:
+
+```ruby
+require 'pagy/extras/headers'
+```
+
+In your controller action:
+
+```ruby
+# paginate as usual with any pagy_* backend constructor
+pagy, records = pagy(Product.all)
+pagy_headers_merge(pagy)
+render json: records
+```
+
+Or if you can reuse it (e.g. in a pure API app with very standard actions), you are encouraged to define a custom `pagy_render` method in your application controller like:
+
+```ruby
+def pagy_render(collection, vars={})
+  pagy, records = pagy(collection, vars) # any pagy_* backend constructor works
+  pagy_headers_merge(pagy)
+  render json: records
+end
```

Comment:
could be more universal to do
```Ruby
def render(*args, &block)
  pagy_headers_merge(@pagy) if @pagy
  super
end
```

Reply:
true! This one is a nice minimalist way to just add the headers automatically that works also with existing code. The other one is more a suggestion about encapsulating/decluttering. I will use both suggestions. Thanks